lighthouse-core/gather/driver.js

```diff
@@ -849,6 +849,26 @@ class Driver {
     });
   }
 
+  async listenForSecurityStateChanges() {
+    this.on('Security.securityStateChanged', state => {
+      this.lastSecurityState = state;
```

Comment:
the type checker now allows this (in JS files as of 3.0 or 3.1), but should still declare it in the constructor so there's a clear place to check for what's on Driver, we don't dynamically mutate the object's shape, etc

Reply:
yeah this really surprised me, how it was able to infer the type. it was able to see that it could be undefined, since the ctor didn't set it.

Makes sense. This is not good:
![image](https://user-images.githubusercontent.com/4071474/47178097-0ba18c80-d2cf-11e8-9271-6340b4ac34af.png)

I guess TSC just infers a permissive union type for `this.property`, based on all setters? What new feature of tsc 3.0/3.1 is this?